### PR TITLE
Implement VarLookup class

### LINKED ISSUE #442: Merge: refactor ids coordinate extraction
Its about this paragraph of code:
https://github.com/duqtools/duqtools/blob/d9a5538b1774338013a4bfa6061fb8bc7a79949d/src/duqtools/ids/_merge.py#L36-L48    

> Perhaps we should even have a var_lookup class that can deal with these cases and transparently pick up $rho_tor_norm and rho_tor_norm as the same variable.

This ties in with what I'm doing for #418 already, so I can take care of that part.

_Originally posted by @stefsmeets in https://github.com/duqtools/duqtools/pull/435#discussion_r1031596298_

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR implements a `VarLookup` class to deal make it a bit more ergonomic to deal with the variable listing. I added support for splitting the `variable.yaml` in multiple files. Duqtools will glob for `variables*.yaml` and merge all variables in these files together in a single list.

Addresses #418 
Closes #442

### CODE REVIEW

File: src/duqtools/ids/_merge.py [hunk 1]
```diff
@@ -29,37 +29,31 @@ def merge_data(
...
-            if not (dim.startswith('$') or dim.startswith('time')):
-                continue
-            name = dim.lstrip('$')
-            if name not in variable_dict:
-                variables.append(var_lookup[name])
-                variable_dict[name] = variables[-1]
+            dim_var = var_lookup[name]
+            variable_dict.setdefault(dim_var.name, dim_var)
+
+    variables = tuple(variable_dict.keys())
```
Comment (stefsmeets) [inline src/duqtools/ids/_merge.py:42]: Hi @v1kko ,
I tried to simplify this code using the new VarLookup class. Can you let me know if I got the intent of this right? 

Essentially, I added all dimensions that were not already a variable to the variable dict / variables listing.

Comment (v1kko) [inline src/duqtools/ids/_merge.py:42]: 1. `keys()` is default
2. should be `values()`
```suggestion
    variables = tuple(variable_dict.values())
```

**Review by v1kko — CHANGES_REQUESTED** [review]
Looks good, testing revealed some errors, but nothing major

File: src/duqtools/list_variables.py [hunk 2]
```diff
@@ -27,7 +25,7 @@ def list_group(group: List, extra_variables: Dict):
         except AttributeError:
             sub = cs(f'({var.type})', **ST_INFO)
 
-        print(f'    - {star}{name} {sub}')
+        click.echo(f'    - {star}{name} {sub}')
```
[the remark this answers is not in the document]
Comment (stefsmeets) [reply src/duqtools/list_variables.py:28]: Log the entire output of `list-variables`?

**Comment by stefsmeets** [comment]
> Looks good, testing revealed some errors, but nothing major

Thanks for testing the PR! Some sloppy errors on my side when resolving the merge conflict 😅

File: src/duqtools/list_variables.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (v1kko) [reply src/duqtools/list_variables.py:28]: okay maybe not

File: src/duqtools/ids/_merge.py [hunk 3]
```diff
@@ -29,37 +29,31 @@ def merge_data(
...
         for dim in variable.dims:
-            # TODO time should also be $ prefixed, remove the `or time` when fixed
-            if not (dim.startswith('$') or dim.startswith('time')):
-                continue
-            name = dim.lstrip('$')
-            if name not in variable_dict:
-                variables.append(var_lookup[name])
-                variable_dict[name] = variables[-1]
+            dim_var = var_lookup[name]
+            variable_dict.setdefault(dim_var.name, dim_var)
```
[the remark this answers is not in the document]
Comment (stefsmeets) [reply src/duqtools/ids/_merge.py]: Ah, good point. Now I see that `ion` is a dimension without coordinates, whereas we want to extract the dimensions *with* coordinates.

[the remark this answers is not in the document]
Comment (stefsmeets) [reply src/duqtools/ids/_merge.py]: fixed

**Pull request merged**